trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/rest/TestAwsChunkedInputStream.java

```diff
@@ -0,0 +1,348 @@
+        InputStream in = new AwsChunkedInputStream(inputStream, new DummyChunkSigningSession());
+        byte[] buffer = new byte[300];
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        int len;
+        while ((len = in.read(buffer)) > 0) {
+            out.write(buffer, 0, len);
+        }
+        assertEquals(-1, in.read(buffer));
+        assertEquals(-1, in.read(buffer));
+
+        in.close();
+
+        String result = out.toString(UTF_8);
+        assertEquals(CHUNKED_RESULT, result);
+    }
+
+    //Test for when buffer is smaller than chunk size.
+    @Test
+    public void testChunkedInputStreamSmallBuffer()
+            throws IOException
+    {
+        ByteArrayInputStream inputStream = new ByteArrayInputStream(CHUNKED_INPUT.getBytes(UTF_8));
+        InputStream in = new AwsChunkedInputStream(inputStream, new DummyChunkSigningSession());
+
+        byte[] buffer = new byte[7];
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        int len;
+        while ((len = in.read(buffer)) > 0) {
+            out.write(buffer, 0, len);
+        }
+        assertEquals(-1, in.read(buffer));
+        assertEquals(-1, in.read(buffer));
+
+        String result = out.toString(UTF_8);
+        assertEquals("123456789012345612345", result);
+    }
+
+    // One byte read
+    @Test
+    public void testChunkedInputStreamOneByteRead()
+            throws IOException
+    {
+        String s = "5;chunk-signature=0\r\n01234\r\n5;chunk-signature=0\r\n56789\r\n0;chunk-signature=0\r\n\r\n";
+        ByteArrayInputStream inputStream = new ByteArrayInputStream(s.getBytes(UTF_8));
+        InputStream in = new AwsChunkedInputStream(inputStream, new DummyChunkSigningSession());
+        int ch;
+        int i = '0';
+        while ((ch = in.read()) != -1) {
+            assertEquals(i, ch);
+            i++;
+        }
+        assertEquals(-1, in.read());
+        assertEquals(-1, in.read());
+
+        in.close();
+    }
+
+    // Missing closing chunk
+    @Test
+    public void testChunkedInputStreamNoClosingChunk()
+    {
+        String s = "5;chunk-signature=0\r\n01234\r\n";
+        ByteArrayInputStream inputStream = new ByteArrayInputStream(s.getBytes(UTF_8));
+        InputStream in = new AwsChunkedInputStream(inputStream, new DummyChunkSigningSession());
+        byte[] tmp = new byte[5];
+        // altered from original test. Our AwsChunkedInputStream is improved and throws when the final chunk is missing or bad
+        assertThrows(IOException.class, () -> in.read(tmp));
+    }
+
+    // Truncated stream (missing closing CRLF)
+    @Test
+    public void testCorruptChunkedInputStreamTruncatedCRLF()
+            throws IOException
+    {
+        // altered to add a few more bad stings
+        Stream.of("5;chunk-signature=0\r\n01234", ";chunk-signature=0\\r\\n01234\\r\\n", "5;chunk-signature=0\\r\\n012340;chunk-signature=0\\r\\n\\r\\n")
```

Comment:
The `\` in `\r` and `\n` should not be escaped in any case

Reply:
Oops - IntelliJ did that.